### ISSUE TITLE
About REUSE_CACHE_IF_EXISTS and max_examples_per_split usage

### ISSUE THREAD

**Issue by ig-perez** [tensorflow/datasets#3513]
**What I need help with / What I was wondering**
Hi, I have downloaded and processed an OPUS en-es dataset using the following snippet:

```python
        build_config = tfds.translate.opus.OpusConfig(
            version=tfds.core.Version("0.1.0"),
            language_pair=("es", "en"),
            subsets=["OpenSubtitles"]
        )
        # OPUS only provides one split: "train"
        builder = tfds.builder("opus", config=build_config)

        # By default stored in /root/tensorflow_datasets
        dwn_config = tfds.download.DownloadConfig(
            extract_dir=dwn_path,  # store extracted files here
            download_mode=tfds.GenerateMode.REUSE_CACHE_IF_EXISTS,  # Reuse downloads, fresh dataset
            max_examples_per_split=num_items  # "train" split only
        )

        builder.download_and_prepare(
            download_dir=dwn_path,
            download_config=dwn_config
        )
```

Initially I downloaded `max_examples_per_split=57000` and now I want to increase that number without re downloading the dataset again. When using `download_mode=tfds.GenerateMode.REUSE_CACHE_IF_EXISTS` I got an error saying that I'm trying to overwrite an existing dataset, which is exactly what I was expecting to happen when using the `REUSE_CACHE_IF_EXISTS` flag as the documentation explains:
![image](https://user-images.githubusercontent.com/33359427/135727410-7e84c93a-2c2e-49d5-8885-aaf317d754f5.png)

Could you please let me know if I'm wrongly understanding the documentation? If so, how can I get a new set of training examples with more/less data from the already downloaded dataset?

**What I've tried so far**
- The `REUSE_CACHE_IF_EXISTS` flag but it didn't help.
- Deleting the prepared dataset. This helps, but involves a manual step.

**Environment information**
(if applicable)
* Operating System: Linux
* Python version: 3.7.9
* `tensorflow-datasets`/`tfds-nightly` version: tensorflow-datasets 4.4.0
* `tensorflow`/`tensorflow-gpu`/`tf-nightly`/`tf-nightly-gpu` version: tensorflow 2.5.1 and 2.6.0

**Comment by Conchylicultor** [comment]
To redownload the dataset, simply remove `~/tensorflow_datasets/opus/`. It will remove the dataset files, but not the cached downloads (in `~/tensorflow_datasets/downloads/`).

Note: If using the https://www.tensorflow.org/datasets/cli, you can use `tfds build --overwrite` instead.

**Comment by ig-perez** [comment]
Thanks @Conchylicultor , yes I'm manually deleting the `opus` folder. Will the feature be available with `REUSE_CACHE_IF_EXISTS` in the future? or that flag has other purpose?

**Comment by Conchylicultor** [comment]
Thank you for flagging this out. I believe this flag is not supported anymore and should be removed. Or do you think we should fix the flag instead ?

**Comment by ig-perez** [comment]
I find it useful during experimentation. Sometimes I download a large dataset and I start training a vanilla model with a small amount of prepared examples, then for parameter tuning I increase the number of samples. When I finish I increase the training samples again.

I can manually delete the folder, but the flag is a much cleaner/better solution from my perspective.

This is my personal workflow, I don't know if there are many people doing this or need the feature. If you ask me, I'd say keep it :)

Have a great weekend!

**Comment by Conchylicultor** [comment]
> I find it useful during experimentation. Sometimes I download a large dataset and I start training a vanilla model with a small amount of prepared examples, then for parameter tuning I increase the number of samples. When I finish I increase the training samples again.

Yes, but isn't the `--overwrite` flag not enough for this use-case ?
I usually experiment with 

```
tfds build my_dataset.py --overwrite --max_examples_per_split=3
```

This does exactly what you want (limit the number of examples and overwrite the existing dataset)

See useful flags in https://www.tensorflow.org/datasets/add_dataset#download_and_prepare_tfds_build, or the https://www.tensorflow.org/datasets/cli doc.

**Comment by ccl-core** [comment]
Hi @ig-perez , this issue should be fixed with commit 757f0ddef296409b59755202eac472111054234a

Thank you for reporting this!

**Comment by ig-perez** [comment]
Awesome, thank you guys!
